lnwallet/channel.go

```diff
@@ -5736,13 +5920,19 @@ func (lc *LightningChannel) InitNextRevocation(revKey *btcec.PublicKey) error {
 //
 // NOTE: It is okay for sourceRef to be nil when unit testing the wallet.
 func (lc *LightningChannel) AddHTLC(htlc *lnwire.UpdateAddHTLC,
-	openKey *models.CircuitKey) (uint64, error) {
+	openKey *models.CircuitKey,
+	optFuncs ...AddHtlcOptions) (uint64, error) {
+
+	opts := defaultaddHtlcOptions()
+	for _, optFunc := range optFuncs {
+		optFunc(opts)
+	}
 
 	lc.Lock()
 	defer lc.Unlock()
 
 	pd := lc.htlcAddDescriptor(htlc, openKey)
-	if err := lc.validateAddHtlc(pd); err != nil {
+	if err := lc.validateAddHtlc(pd, opts.buffer); err != nil {
```

Comment:
looks like `AddHTLC` will only be called with `FeeBuffer` option, so I guess the `AddHtlcOptions` can be removed to simply the code?

Reply:
The reason I introduced the variadic parameter is, if I would add a normal function parameter for AddHTLC, I would need to edit all the tests where `AddHTLC` is used (a lot) and would need to add a function parameter although none of them use a buffer anyways. But also happy to do the work and change if you think that's a cleaner approach.

Reply:
I agree, I think we should get rid of the variadic param and do the work and make sure all testcases which are not BOLT03 testcases adhere to the buffer.

Reply:
I came up with a slightly different approach, I enhanced the method with an unexported method `addHTLC` which is needed to test all the scenarios e.g. BOLT03 test vectors and kept the `AddHTLC` function signature and always enforcing the feebufffer in it, lmkwyt ?